contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -2088,6 +2089,292 @@ numeric_radians(PG_FUNCTION_ARGS)
+                {
+                    end_positions[current_part] = &object_name_str[i - 1];
+                }
+
+                current_part++;
+                if (current_part > 3)
+                {
+                    PG_RETURN_NULL();
+                }
+
+                start_positions[current_part] = &object_name_str[i + 1];
+                total_chars = 0;
+                total_length = 0;
+            }
+        }
+        else if (state == 1)
+        {
+            if (c == '"')
+            {
+                // is there a next character and is it double quotes?
+                if (i + consumed < len && object_name_str[i + consumed] == '"')
+                {
+                    i += consumed;
+                }
+                else
+                {
+                    state = 0;
+                    end_positions[current_part] = &object_name_str[i - 1];
+                    if (i + 1 < len && object_name_str[i + 1] != '.')
+                    {
+                        PG_RETURN_NULL();
+                    }
+                    i += consumed;
+                    continue;
+                }
+            }
+        }
+        else if (state == 2)
+        {
+            if (c == ']')
+            {
+                // is there a next character and if it is there, is it closing brace?
+                if (i + consumed < len && object_name_str[i + consumed] == ']')
+                {
+                    i += consumed;
+                }
+                else
+                {
+                    state = 0;
+                    end_positions[current_part] = &object_name_str[i - 1];
+                    if (i + 1 < len && object_name_str[i + 1] != '.')
+                    {
+                        PG_RETURN_NULL();
+                    }
+                    i += consumed;
+                    continue;
+                }
+            }
+        }
+
+        // just add the total characters and total length
+        if (state > 0 || (state == 0 && c != '.'))
+        {
+            if (code <= 0xFFFF)
+                total_chars += 1;
+            else
+                total_chars += 2;
+            total_length += (code <= 0xFFFF) ? 2 : 4;
+        }
+        i += consumed;
+    }
+
+    if (state != 0)
+    {
+        PG_RETURN_NULL();
+    }
+
+    if (total_chars > 128 || total_length > 256)
+    {
+        PG_RETURN_NULL();
+    }
+
+    // if there is only 1 part and no '.', set the end position to length-1.
+    if (end_positions[current_part] == NULL)
+    {
+        end_positions[current_part] = &object_name_str[len - 1];
+    }
+
+    // Reverse the object piece index
+    object_piece = current_part + 1 - object_piece;
+    if (object_piece < 0 || object_piece > current_part)
+    {
+        PG_RETURN_NULL();
+    }
+    
+    if (object_piece >= 0 && object_piece <= current_part)
```

Comment:
Already did the check, can omit this

Reply:
I agree with the original comment, due to the check on line 2364, is the following if statement not guaranteed to always be true? Since it can't be false, it the check should be removed.